pkg/executor/cluster.go

```diff
@@ -212,22 +70,10 @@ func addFinalizer(cr *kdv1.KubeDirectorCluster) bool {
 		}
 	}
 	if found {
-		return false
+		return nil
 	}
 	cr.Finalizers = append(cr.Finalizers, finalizerID)
-	return true
-}
-
-func removeFinalizer(cr *kdv1.KubeDirectorCluster) bool {
-	found := false
-	for i, f := range cr.Finalizers {
-		if f == finalizerID {
-			cr.Finalizers = append(cr.Finalizers[:i], cr.Finalizers[i+1:]...)
-			found = true
-			break
-		}
-	}
-	return found
+	return shared.Client().Update(context.TODO(), cr)
```

Comment:
Should have the comment here about moving to PATCH when available.

Reply:
Done.